core/components/broker/broker.go

```diff
@@ -245,18 +246,27 @@ func (b component) HandleData(bctx context.Context, req *core.DataBrokerReq) (*c
 	fhdr := &uplinkPayload.MACPayload.(*lorawan.MACPayload).FHDR // No nil ref, ensured by NewLoRaWANData()
 	fcnt16 := fhdr.FCnt                                          // Keep a reference to the original counter
 
+	var fcntReset bool
 	var mEntry *devEntry
 	for _, entry := range entries {
+		fcntReset = false
```

Comment:
We're inside a loop at that point. The counter condition can set the value to true where MIC validation may continue to the next element where the variable needs to be reset.